test/utils.py

```diff
@@ -260,25 +260,63 @@ def _save(self, p):
         self.outputs.append(p.outputs.copy())
 
 
-class ProcessSaver(plumpy.ProcessListener, Saver):
+_ProcessSaverProcReferences = {}
+_ProcessSaver_Saver = {}
+
+
+class ProcessSaver(plumpy.ProcessListener):
```

Comment:
Can you please add to the docstring of this class why these manual mappings of process and saver references are necessary and how it works.